src/loadsave.cpp

```diff
@@ -109,7 +109,22 @@ char				sRequestResult[PATH_MAX];   // filename returned;
 bool				bRequestLoad = false;
 LOADSAVE_MODE		bLoadSaveMode;
 static const char *savedTitle;
-static const char *sExt = ".gam";
+static const char *sSaveGameExtension = ".gam";
+
+// ////////////////////////////////////////////////////////////////////////////
+// return whether the specified filename looks like a saved game file, i.e. ends with .gam
+bool isASavedGamefile(const char* filename)
+{
+	static const size_t saveGameExtensionLength = strlen(sSaveGameExtension);
+
+	if (nullptr == filename)
+	{
+		return false;
+	}
+
+	return 0 == strcmp(filename + strlen(filename) - saveGameExtensionLength, sSaveGameExtension);
```

Comment:
There's a problem if `strlen(filename) < saveGameExtensionLength`

Suggested change:
```suggestion
	size_t filenameLength = strlen(filename);
	if (filenameLength <= saveGameExtensionLength)
	{
		// reject filename of insufficient length to contain "<anything>.gam"
		return false;
	}
	return 0 == strcmp(filename + filenameLength - saveGameExtensionLength, sSaveGameExtension);
```

Reply:
Ah, good spot, thanks.  Apologies, I should have considered that case.